src/libraries/Microsoft.Extensions.Primitives/src/ChangeToken.cs

```diff
@@ -100,7 +100,7 @@ private void RegisterChangeTokenCallback(IChangeToken token)
                     return;
                 }
 
-                IDisposable registraton = token.RegisterChangeCallback(s => ((ChangeTokenRegistration<TState>)s).OnChangeTokenFired(), this);
+                IDisposable registraton = token.RegisterChangeCallback(s => ((ChangeTokenRegistration<TState>?)s)?.OnChangeTokenFired(), this);
```

Comment:
Similar comment here as in `CompositeChangeToken.cs` - this is adding a null check where it isn't necessary. `this` will be passed to this method, so `s` will never be `null`. We can add a `!` here to silence the warning.